src/results.rs

```diff
@@ -0,0 +1,33 @@
+use std::fmt;
+use std::error::Error;
+
+#[derive(Debug)]
+pub struct ParsingError {
+    desc: String,
```

Comment:
Totally agree. I left it that way for now because there were other things to worry about.